### ISSUE TITLE
Extend `Style/MethodCallParentheses` to police method calls with arguments

### ISSUE THREAD

**Issue by Drenmi** [bbatsov/rubocop#3737]
So far, this cop just checks that we're not using redundant parentheses for method calls without arguments. We should extend this to allow enforcing parentheses for method calls with arguments.

This would need to exclude method calls made from within a constant body (e.g. `attr_reader`) and common "keyword" methods (e.g. `puts`), and potentially be configurable to always omit parentheses. (The last part is a bit harder, since sometimes that produces ambiguous code.)

**Example:**

```
foo :bar, :baz
```

should generate an offense and be corrected to:

```
foo(:bar, :baz)
```
